newm-server/src/main/kotlin/io/newm/server/features/marketplace/repo/MarketplaceRepositoryImpl.kt

```diff
@@ -45,9 +46,9 @@ internal class MarketplaceRepositoryImpl(
 
     override suspend fun getSale(saleId: UUID): Sale {
         log.debug { "get: saleId = $saleId" }
-        return transaction {
-            MarketplaceSaleEntity[saleId].toModel()
-        }
+        val sale = transaction { MarketplaceSaleEntity[saleId] }
```

Comment:
Wrap this entire function in one transaction instead of using multiple db transactions.

Reply:
Needs to be separate transactions because we call a suspend function in line 50.